src/health.h

```diff
@@ -111,262 +129,405 @@ typedef struct rrddimvar {
+    uint32_t hash_chart; ///< hash of chart
 
-    char *family;
+    char *family; ///< link to family
 
-    char *exec;
-    char *recipient;
-    time_t exec_run_timestamp;
-    int exec_code;
+    char *exec;                ///< Script to run when the alarm switches state.
+    char *recipient;           ///< First parameter passed to `exec`
+    time_t exec_run_timestamp; ///< timestamp when `exec` run
+    int exec_code;             ///< exit code of `exec`
 
-    char *source;
-    char *units;
-    char *info;
+    char *source; ///< source description of the alarm that generated this alarm. (i.e. `LINE@FILE`)
+    char *units;  ///< how to interpret the values
+    char *info;   ///< description of the alarm
 
-    calculated_number old_value;
-    calculated_number new_value;
+    calculated_number old_value; ///< value before
+    calculated_number new_value; ///< value after
 
-    char *old_value_string;
-    char *new_value_string;
+    char *old_value_string; ///< string representation of `old_value`
+    char *new_value_string; ///< string representation of `new_value`
 
-    int old_status;
-    int new_status;
+    int old_status; ///< integer representation of `old_value`
+    int new_status; ///< integer representation of `new_value`
 
-    uint32_t flags;
+    uint32_t flags; ///< HEALTH_ENTRY_FLAG_*
 
-    int delay;
-    time_t delay_up_to_timestamp;
+    int delay;                    ///< "delay to run the script" as was calculated when the alarm entry was created.
+    time_t delay_up_to_timestamp; ///< "delay to run the script" expressed as an absolute timestamp.
 
-    uint32_t updated_by_id;
-    uint32_t updates_id;
+    uint32_t updated_by_id; ///< id of the alarm entry that updated this alarm entry
+    uint32_t updates_id;    ///< \brief id of the alarm entry that has been updated by this alarm entry.
+                            ///< Tthe one that was cancelled because of this alarm entry.
     
-    struct alarm_entry *next;
+    struct alarm_entry *next; ///< next entry in the list
 } ALARM_ENTRY;
 
+/** Log of alarms */
 typedef struct alarm_log {
-    uint32_t next_log_id;
-    uint32_t next_alarm_id;
-    unsigned int count;
-    unsigned int max;
-    ALARM_ENTRY *alarms;
-    pthread_rwlock_t alarm_log_rwlock;
+    uint32_t next_log_id;   ///< next ID of log
+    uint32_t next_alarm_id; ///< next alarm
+    unsigned int count;     ///< number appeard
+    unsigned int max;       ///< maximal value
+    ALARM_ENTRY *alarms;    ///< alarms which are logged
+    pthread_rwlock_t alarm_log_rwlock; ///< lock to synchronize access to this
 } ALARM_LOG;
 
 #include "rrd.h"
 
+/**
+ * Rename all variables for RRDSET `st`.
+ *
+ * @param st RRDSET to rename variables for.
+ */
 extern void rrdsetvar_rename_all(RRDSET *st);
+/**
+ * Create a variable for RRDSET `st`.
+ *
+ * @param st RRDSET to create RRDSETVAR for.
+ * @param variable name.
+ * @param type RRDVAR_TYPE_*
+ * @param value ktsaou: Your help needed.
```

Comment:
RRDVAR has a `void *value`. RRDVAR_TYPE_* state the type of data the pointer points to.